apis/cassandra/v1beta1/cassandradatacenter_types.go

```diff
@@ -188,9 +188,12 @@ type CassandraDatacenterSpec struct {
 	// If it is omitted, we will generate a secret instead.
 	SuperuserSecretName string `json:"superuserSecretName,omitempty"`
 
-	// The k8s service account to use for the server pods
+	// DEPRECATED Use ServiceAccountName instead, which takes precedence. The k8s service account to use for the server pods
 	ServiceAccount string `json:"serviceAccount,omitempty"`
```

Comment:
nit: I noticed that in the Kubernetes code they renamed the Go field to `DeprecatedServiceAccount`, seems like a useful convention.

Reply:
We can try that, it'll probably cause some issues for someone upgrading - but the YAML stays intact.